.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/ChineseDurationParserConfiguration.cs

```diff
@@ -33,33 +45,36 @@ public DateTimeParseResult Parse(ExtractResult er, DateObject refDate)
         {
             var referenceTime = refDate;
 
-            var parseResult = InternalParser.Parse(er);
-            var unitResult = parseResult.Value as UnitValue;
-
-            if (unitResult == null)
+            var dateTimeParseResult = ParseMergedDuration(er.Text, referenceTime);
+            if (!dateTimeParseResult.Success)
             {
-                return null;
-            }
+                var parseResult = InternalParser.Parse(er);
+                var unitResult = parseResult.Value as UnitValue;
 
-            var dateTimeParseResult = new DateTimeResolutionResult();
-            var unitStr = unitResult.Unit;
-            var numStr = unitResult.Number;
+                if (unitResult == null)
+                {
+                    return null;
+                }
 
-            dateTimeParseResult.Timex = "P" + (BaseDurationParser.IsLessThanDay(unitStr) ? "T" : string.Empty) + numStr + unitStr[0];
-            dateTimeParseResult.FutureValue = dateTimeParseResult.PastValue = double.Parse(numStr) * UnitValueMap[unitStr];
-            dateTimeParseResult.Success = true;
+                var unitStr = unitResult.Unit;
+                var numStr = unitResult.Number;
+
+                dateTimeParseResult.Timex = "P" + (BaseDurationParser.IsLessThanDay(unitStr) ? "T" : string.Empty) + numStr + unitStr[0];

```

Comment:
As you're touching this code, can we also try to refactor it so timex generation goes to a "centralized" module?

Reply:
OK, I will try to modify it.